parsers/ENTSOE.py

```diff
@@ -817,6 +836,76 @@ def parse_prices(
     return prices
 
 
+def parse_outages(
+    xml_trees: list[ET.ElementTree],
+    zoneKey: ZoneKey,
+    logger: Logger,
+) -> OutageList:
+    if not xml_trees:
+        return OutageList(logger)
+    outages = OutageList(logger)
+    for tree in xml_trees:
+        xml_string = ET.tostring(tree.getroot(), encoding="unicode")
+        xml_string_without_ns = re.sub(r"ns\d+:", "", xml_string)
+        soup = BeautifulSoup(xml_string_without_ns, "xml")
+        reason = soup.find("Reason").find("text").contents[0]
+        for timeseries in soup.find_all("TimeSeries"):
+            fuel_code = str(
+                timeseries.find(
+                    "production_RegisteredResource.pSRType.psrType"
+                ).contents[0]
+            )
+            fuel_em_type = ENTSOE_PARAMETER_BY_GROUP[fuel_code]
+            outage_type = OutageType.mapping_code_to_type(
+                timeseries.find("businessType").contents[0]
+            )
+            generator_id = str(
+                timeseries.find("production_RegisteredResource.mRID").contents[0]
+            )
+            installed_capacity = float(
+                timeseries.find(
+                    "production_RegisteredResource.pSRType.powerSystemResources.nominalP"
+                ).contents[0]
+            )
+
+            for entry in timeseries.find_all("Available_Period"):
+                quantity = float(entry.find("Point").find("quantity").contents[0])
+                capacity_reduction = installed_capacity - quantity
+
+                time_range = entry.find("timeInterval")
+                start_time = time_range.find("start").contents[0]
+                end_time = time_range.find("end").contents[0]
+                datetime_start = datetime.fromisoformat(zulu_to_utc(f"{start_time}"))
+                datetime_start_rounded = datetime_start.replace(
+                    minute=0, second=0, microsecond=0
+                ) + timedelta(hours=1)
+                datetime_end = datetime.fromisoformat(
+                    zulu_to_utc(f"{end_time}")
+                ).replace(minute=0, second=0, microsecond=0) + timedelta(
+                    hours=1
+                )  # round to the next hour\
+
+                # HACK: creating one datetime per hour but should rather have one event per outage and handle this downstream.
+                for dt in [
+                    datetime_start,
+                    *list(
+                        pd.date_range(datetime_start_rounded, datetime_end, freq="H")
```

Comment:
What's the benefit of this vs the existing datetime parsing in the ENTSOE parser?

This is also not guaranteed to be hourly and could be 15 minute intervals I think.